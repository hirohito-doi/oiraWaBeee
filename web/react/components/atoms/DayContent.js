import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { CardActionArea, CardContent, Typography, Avatar, Badge } from '@material-ui/core';
import moment from 'moment';

const useStyles = makeStyles(theme => ({
  sunday: {
    color: "#c51162"
  },
  satruday: {
    color: "#303f9f"
  },
  day: {
    width: 100,
    height: 100,
    borderRight: "solid 1px #ddd",
    '&:last-child': {
      borderRight: 'none',
    }
  },
  dayBefore: {
    opacity: .4
  },
  activeMessage: {
    minHeight: 75,
    padding: 10
  },
  avatar: {
    width: 25,
    height: 25
  },
  badge: {
    top: 5,
  },
  message: {
    position: "relative",
    padding: 1,
    marginTop: 7,
    border: "solid 1px #ccc",
    borderRadius: 2,
    '&::before': {
      content: '""',
      position: "absolute",
      top: -18,
      left: 3,
      border: "9px solid transparent",
      borderBottom: "9px solid #ddd",
      borderLeft: "8px solid transparent",
      borderRight: "8px solid transparent"
    },
    '&:after': {
      content: '""',
      position: "absolute",
      top: -16,
      left: 5,
      border: "8px solid transparent",
      borderBottom: "8px solid #fff",
      borderLeft: "6px solid transparent",
      borderRight: "6px solid transparent"
    },
  }
}));

/**
 * 日表示
 * @param {*} props
 */
const DayContent = (props) => {
  const { year, month, day, dateSelects, toggleDateSelect, index } = props;
  const classes = useStyles();

  const m = moment({y:year, M:month, d:day});
  const now = moment();

  // 曜日表示の色分け
  let classDayColor = "";
  if (index==0) { // 日曜日
    classDayColor += ` ${classes.sunday}`;
  } 
  if (index==6) { // 土曜日
    classDayColor += ` ${classes.satruday}`;
  }

  // 今日より前の日付か
  let classDay = `${classes.day}`;
  let handleClick = ()=>{toggleDateSelect(m.format('YYYY-MM-DD'))}
  if(m.isBefore(now, 'day')) {
    classDay += ` ${classes.dayBefore}`
    handleClick = ()=>{}
  }

  // 選択済みチェック
  let isSelected = false;
  if(!m.isBefore(now, 'day') && dateSelects.indexOf(m.format('YYYY-MM-DD')) >= 0) {
    isSelected = true;
  }

  return (
    <CardActionArea
      className={classDay}
      onClick={handleClick}
    >
      <Typography className={classDayColor}>
        <Badge
          className={classes.badge}
          color="secondary"
          variant="dot"
          invisible={!m.isSame(now, 'day')}
        >
          {day}
        </Badge>
      </Typography>
      <CardContent className={classes.activeMessage}>
        {day && isSelected &&
          <React.Fragment>
            <Avatar className={classes.avatar} alt="Beeee" src="/static/beeee.jpg" />
            <Typography className={classes.message} variant="caption" display="block">
              まかせろぁ！
            </Typography>
          </React.Fragment>
        }
      </CardContent>
    </CardActionArea>
  )
}

export default DayContent;