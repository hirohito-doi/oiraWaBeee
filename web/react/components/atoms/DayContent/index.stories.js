import React from 'react';
import { action } from '@storybook/addon-actions';
import moment from 'moment';

import DayContent from '.'

export default {
  component: DayContent,
  title: 'DayContent',
};

export const Today = () => {
  const m = moment();
  return (
    <DayContent
      year={m.year()}
      month={m.month()}
      day={m.date()}
      dateSelects={[]}
      toggleDateSelect ={action('clicked')}
      index={m.day()}
    />
  )
};

export const TodayOn = () => {
  const m = moment();
  return (
    <DayContent
      year={m.year()}
      month={m.month()}
      day={m.date()}
      dateSelects={[m.format("YYYY-MM-DD")]}
      toggleDateSelect ={action('clicked')}
      index={m.day()}
    />
  )
};

export const Yesterday = () => {
  const m = moment().subtract(1, 'days');
  return (
    <DayContent
      year={m.year()}
      month={m.month()}
      day={m.date()}
      dateSelects={[m.format("YYYY-MM-DD")]}
      toggleDateSelect ={action('clicked')}
      index={m.day()}
    />
  )
};