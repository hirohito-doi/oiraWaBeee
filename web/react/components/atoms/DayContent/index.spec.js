import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import moment from 'moment';

import DayContent from '.';

describe('DayContent', () => {
  it('render correctory', () => {
    const tree = renderer
      .create(<DayContent
        year={2999}
        month={10}
        day={10}
        dateSelects={[]}
        toggleDateSelect ={jest.fn()}
        index={0}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  })

  it('check selected', () => {
    const onClick = jest.fn();
    const m = moment();

    const tree = mount(<DayContent
      changeMonth={onClick}
      year={m.year()}
      month={m.month()}
      day={m.date()}
      dateSelects={[m.format("YYYY-MM-DD")]}
      toggleDateSelect ={onClick}
      index={m.day()}
    />)

    // チェック済テスト
    expect(tree.hasClass('MuiAvatar-img')).toEqual(true);
    expect(tree.text()).toMatch('まかせろぁ！');

    // クリックテスト
    tree.find('button').simulate('click');
    expect(onClick).toHaveBeenCalled();
  })
})