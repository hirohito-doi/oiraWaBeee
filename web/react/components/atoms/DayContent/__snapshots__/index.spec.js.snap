// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DayContent render correctory 1`] = `
<button
  className="MuiButtonBase-root MuiCardActionArea-root makeStyles-day-3"
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <p
    className="MuiTypography-root  makeStyles-sunday-1 MuiTypography-body1"
  >
    <span
      className="MuiBadge-root makeStyles-badge-7"
    >
      10
      <span
        className="MuiBadge-badge MuiBadge-colorSecondary MuiBadge-invisible MuiBadge-dot"
      >
        
      </span>
    </span>
  </p>
  <div
    className="MuiCardContent-root makeStyles-activeMessage-5"
  />
  <span
    className="MuiCardActionArea-focusHighlight"
  />
</button>
`;
